File: TeachMeAPI/TeachMeAPI/Models/Announcement.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Linq;
using System.Web;

using TeachMeAPI.DAO;

namespace TeachMeAPI.Models
{
    public class Announcement
    {
        [JsonProperty(PropertyName = "idAnnouncement")]
        public int IdAnnouncement { get; set; }
        [JsonProperty(PropertyName = "idTutor")]
        public int IdTutor { get; set; }
        [JsonProperty(PropertyName = "idCourse")]
        public int IdCourse { get; set; }
        [JsonProperty(PropertyName = "title")]
        public string Title { get; set; }
        [JsonProperty(PropertyName = "description")]
        public string Description { get; set; }
        [JsonProperty(PropertyName = "fee")]
        public double Fee { get; set; }

        public Announcement()
        { }

        public Announcement(int idAnnouncement, int idTutor, int idCourse, string title, string description, double fee)
        {
            IdAnnouncement = idAnnouncement;
            IdCourse = idCourse;
            IdTutor = idTutor;
            Title = title;
            Description = description;
            Fee = fee;
        }

        public Announcement(SqlDataReader reader)
        {
            IdAnnouncement = Int32.Parse(reader[AnnouncementDAO.COLUMN_ID_ANNOUNCEMENT].ToString());
            IdCourse = Int32.Parse(reader[AnnouncementDAO.COLUMN_ID_COURSE].ToString());
            IdTutor = Int32.Parse(reader[AnnouncementDAO.COLUMN_ID_TUTOR].ToString());
            Title = reader[AnnouncementDAO.COLUMN_TITLE].ToString();
            Description = reader[AnnouncementDAO.COLUMN_DESCRIPTION].ToString();
            Fee = Double.Parse(reader[AnnouncementDAO.COLUMN_FEE].ToString());
        }
    }
}

File: TeachMeWeb/src/app/announcement/announcement.ts
import {Course} from "../course/course";
import {Tutor} from "../user/tutor";

export class Announcement{

  private _idAnnouncement:number;
  private _title:string;
  private _description:string;
  private _fee:number;
  private _idCourse: number
    private _idTutor:number;

  constructor(title: string="", description: string="", fee: number=0, course: number=-1, tutor: number=-1){
    this._title=title;
    this._description=description;
    this._fee=fee;
    this._idCourse=course;
    this._idTutor=tutor;
  }

  get idTutor(): number {
    return this._idTutor;
  }

  set idTutor(value: number) {
    this._idTutor = value;
  }
  get idCourse(): number {
    return this._idCourse;
  }

  set idCourse(value: number) {
    this._idCourse = value;
  }
  get idAnnouncement(): number {
    return this._idAnnouncement;
  }

  set idAnnouncement(value: number) {
    this._idAnnouncement = value;
  }

  get fee(): number {
    return this._fee;
  }

  set fee(value: number) {
    this._fee = value;
  }
  get description(): string {
    return this._description;
  }

  set description(value: string) {
    this._description = value;
  }
  get title(): string {
    return this._title;
  }

  set title(value: string) {
    this._title = value;
  }

  public serialize() : any {
    return {
      idAnnouncement: this._idAnnouncement,
      idTutor: this._idTutor,
      idCourse: this._idCourse,
      title: this._title,
      description: this._description,
      fee: this._fee
    };
  }

  public deserializable(json: any) : Announcement {
    Object.assign(this, json);
    return this;
  }


}
